smartsim/_core/control/controller.py

```diff
@@ -625,39 +661,63 @@ def _set_dbobjects(self, manifest: Manifest) -> None:
         if not manifest.has_db_objects:
             return
 
-        db_addresses = self._jobs.get_db_host_addresses()
+        address_dict = self._jobs.get_db_host_addresses()
+        # db_address, address_dict = self._jobs.get_db_host_addresses()
+        flag = ""
+        for (db_id,db_addresses,) in address_dict.items():
+            db_name = "_".join(db_id.split("_")[:-1])
 
-        hosts = list({address.split(":")[0] for address in db_addresses})
-        ports = list({int(address.split(":")[-1]) for address in db_addresses})
+            if db_name == "orchestrator":
+                db_name = ""
+                flag = "blank"
 
-        if not db_is_active(hosts=hosts, ports=ports, num_shards=len(db_addresses)):
-            raise SSInternalError("Cannot set DB Objects, DB is not running")
+            db_without = db_name
+            if not flag == "blank":
+                db_name = "_" + db_name
 
-        client = Client(address=db_addresses[0], cluster=len(db_addresses) > 1)
+            hosts = list({address.split(":")[0] for address in db_addresses})
+            ports = list({int(address.split(":")[-1]) for address in db_addresses})
 
-        for model in manifest.models:
-            if not model.colocated:
-                for db_model in model.db_models:
+            if not db_is_active(hosts=hosts, ports=ports, num_shards=len(db_addresses)):
+                raise SSInternalError("Cannot set DB Objects, DB is not running")
```

Comment:
Should we replicate this call to other places?

Reply:
Where else?